Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.cpp

```diff
@@ -0,0 +1,543 @@
+
+            bool HairFeatureProcessor::InitPPLLResolvePass()
+            {
+                m_hairPPLLResolvePass = nullptr;   // reset it to null, just in case it fails to load the assets properly
+
+                RPI::PassSystemInterface* passSystem = RPI::PassSystemInterface::Get();
+                if (passSystem->HasPassesForTemplateName(AZ::Name{HairPPLLResolvePassTemplateName}))
+                {
+                    auto& desiredPasses = passSystem->GetPassesForTemplateName(AZ::Name{HairPPLLResolvePassTemplateName});
+
+                    // For now, assume one skinning pass
+                    if (!desiredPasses.empty() && desiredPasses[0])
+                    {
+                         m_hairPPLLResolvePass = static_cast<HairPPLLResolvePass*>(desiredPasses[0]);
+                         m_hairPPLLResolvePass->SetFeatureProcessor(this);
+                    }
+                    else
+                    {
+                        AZ_Error("Hair Gem", false, "HairPPLLResolvePassTemplate does not have any valid passes. Check your game project's .pass assets.");
+                        return false;
+                    }
+                }
+                else
+                {
+                    AZ_Error("Hair Gem", false, "Failed to find passes for HairPPLLResolvePassTemplate. Check your game project's .pass assets.");
+                    return false;
+                }
+                return true;
+            }
+
+            void HairFeatureProcessor::BuildDispatchAndDrawItems(Data::Instance<HairRenderObject> renderObject)
+            {
+                HairRenderObject* renderObjectPtr = renderObject.get();
+
+                // Dispatches for Compute passes
+                m_computePasses[GlobalShapeConstraintsPass]->BuildDispatchItem(
+                    renderObjectPtr, DispatchLevel::DISPATCHLEVEL_VERTEX);
+                m_computePasses[CalculateStrandDataPass]->BuildDispatchItem(
+                    renderObjectPtr, DispatchLevel::DISPATCHLEVEL_STRAND);
+                m_computePasses[VelocityShockPropagationPass]->BuildDispatchItem(
+                    renderObjectPtr, DispatchLevel::DISPATCHLEVEL_VERTEX);
+                m_computePasses[LocalShapeConstraintsPass]->BuildDispatchItem(
+                    renderObjectPtr, DispatchLevel::DISPATCHLEVEL_STRAND);
+                m_computePasses[LengthConstriantsWindAndCollisionPass]->BuildDispatchItem(
+                    renderObjectPtr, DispatchLevel::DISPATCHLEVEL_VERTEX);
+                m_computePasses[UpdateFollowHairPass]->BuildDispatchItem(
+                    renderObjectPtr, DispatchLevel::DISPATCHLEVEL_VERTEX);
+
+                // Render / Raster pass - adding the object will schedule Srgs binding
+                // and DrawItem build.
+                m_hairPPLLRasterPass->SchedulePacketBuild(renderObjectPtr);
+            }
+
+            Data::Instance<HairSkinningComputePass> HairFeatureProcessor::GetHairSkinningComputegPass()
```

Comment:
While it might look this way, it is not the case.
All skinning compute passes are identical (share algorithm, class and shader), and so I take the first one in the array.
If you track the usage you'd see that this is done to create the generic compute dispatch for this pass type.